### ISSUE TITLE
lr_scheduler.step have undesired result when used on multi gpu.

### ISSUE THREAD

**Issue by crj1998** [huggingface/accelerate#2142]
### System Info

```Shell
Copy-and-paste the text below in your GitHub issue

- `Accelerate` version: 0.24.1
- Platform: Linux-5.15.133+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Numpy version: 1.24.3
- PyTorch version (GPU?): 2.0.0 (True)
- PyTorch XPU available: False
- PyTorch NPU available: False
- System RAM: 31.36 GB
- GPU type: Tesla T4
- `Accelerate` default config:
        Not found
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

When i use a lr scheduler, I want to change lr according to steps. The follow figure is when my code run single gpu.
![save3](https://github.com/huggingface/accelerate/assets/28671582/ce353369-287e-4b81-96a6-963d5e8b9daa)

Hower when i run the following scripts use `accelerate launch --multi_gpu --main_process_port 12345 debug.py` on a two gpu node. the lr curve is:
![save2](https://github.com/huggingface/accelerate/assets/28671582/d39209f4-961e-45c7-9326-e23205579d31)

Obviously the step function was updated twice, which I think is probably because the `current step` is thread-unsafe.

so when i set `global step` i.e. `lr_scheduler.step(global_step )`, the result is desired.

![save1](https://github.com/huggingface/accelerate/assets/28671582/f22909d9-e81c-4b79-8452-c1a341b288dd)

A minimal scripts for reproduce

```python
from accelerate import Accelerator

import math
import torch
import torch.nn as nn
from torch.optim.lr_scheduler import LambdaLR

accelerator = Accelerator()

model = nn.Linear(2, 2)
optimizer = torch.optim.AdamW(model.parameters(), lr=0.1)
num_warmup_steps = 20
num_training_steps = 100

from transformers import get_linear_schedule_with_warmup
lr_scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps=num_warmup_steps, num_training_steps=num_training_steps)

model, optimizer, lr_scheduler = accelerator.prepare(model, optimizer, lr_scheduler)

global_step = 0
lrs = []
while global_step < num_training_steps:
    x = torch.randn(2, 2).to(accelerator.device)
    y = torch.randn(2, 2).to(accelerator.device)
    loss = torch.nn.functional.mse_loss(model(x), y)
    accelerator.backward(loss)
    optimizer.step()
    lr_scheduler.step()
    optimizer.zero_grad()

    if accelerator.sync_gradients:
        lr = lr_scheduler.get_last_lr()[0]
        lrs.append(lr)
        accelerator.print(global_step, lr)
        global_step += 1

if accelerator.is_main_process:
    import matplotlib.pyplot as plt
    plt.style.use("ggplot")
    plt.xlabel("steps")
    plt.ylabel("lr")
    plt.plot(lrs)
    plt.title("accelerate launch --multi_gpu")
    plt.savefig("save.png")
```

### Expected behavior

Generally, we use `scheduler.step()` directly , rather than pass a `global_step`, i think this should be State clearly. I am not this is a feature or a bug.

**Comment by qixuema** [comment]
Hello @crj1998 ,

Happy New Year! 🎉🎉🎉

I also encountered a similar issue while using `accelerate` for multi-GPU training, where the scheduler seems to be updated multiple times. Apart from setting `global_step`, are there any other methods to address this? Thank you very much for sharing!

Xueqi

**Comment by qixuema** [comment]
@muellerzr Thank you!

**Comment by donthomasitos** [comment]
@muellerzr Thanks for clarifying. I think this is unintuitive and deserves documentation, maybe at the parameter desc of `class Accelerator`.

**Comment by muellerzr** [comment]
Sure, I can look into that! For now it is mentioned here: https://huggingface.co/docs/accelerate/concept_guides/performance

**Comment by donthomasitos** [comment]
Great! Thanks. As as side note, maybe there's a bug: I observed the learning rate scheduler to be twice as fast (as if it's two scheduler.step() per training batch) but I'm working with 8 GPUs and 8 processes. With `step_scheduler_with_optimizer=False` it works as expected with only one scheduler.step() per training batch. So I would have expected the difference to be 8x not 2x.

**Comment by muellerzr** [comment]
@donthomasitos hows your Accelerator configured fully?
